### PR TITLE
Proper fix for formula fetching cells from grid

### PR DESCRIPTION
**Pull request by HactarCE**
This is a proper fix for #1716 plus a test to ensure this doesn't happen again

### CODE REVIEW

File: quadratic-core/src/controller/mod.rs [hunk 1]
```diff
@@ -65,7 +65,8 @@ impl GridController {
             .map(|t| &t.forward)
     }
 
-    pub fn test_blank() -> Self {
+    #[cfg(test)]
```
Comment (ddimaria) [inline quadratic-core/src/controller/mod.rs:68]: 🙌

File: quadratic-core/src/grid/sheet.rs [hunk 2]
```diff
@@ -241,7 +241,21 @@ impl Sheet {
...
     pub fn get_cell_for_formula(&self, pos: Pos) -> CellValue {
-        self.display_value(pos).unwrap_or(CellValue::Blank)
+        let cell_value = self
+            .get_column(pos.x)
+            .and_then(|column| column.values.get(&pos.y));
+
+        if let Some(cell_value) = cell_value {
+            match cell_value {
+                CellValue::Blank | CellValue::Code(_) => match self.code_runs.get(&pos) {
+                    Some(run) => run.get_cell_for_formula(0, 0),
```
Comment (ddimaria) [inline quadratic-core/src/grid/sheet.rs:251]: Out of curiosity, why is this set to `(0. 0)`?

**Review by ddimaria — APPROVED** [review]

**Review by davidkircos — APPROVED** [review]
Lgtm! Cool to see errors moving correctly from Python > Formulas, checkable with IFERROR

File: quadratic-core/src/grid/sheet.rs (hunk 2 above)
Comment (HactarCE) [reply quadratic-core/src/grid/sheet.rs:251]: If there is a code cell _at_ the given location, then we just need the very first value in its output

**Pull request merged**